CalPlayer/Dashes/PlayerDashManager.cs

```diff
@@ -13,6 +14,32 @@ public static bool FindByID(string id, out PlayerDashEffect dashEffect)
             return DashIdentificationTable.TryGetValue(id, out dashEffect);
         }
 
+        // A new function that allows to add in new modded dashes for Calamity to recognize, it attempts
+        // to make it as safe as possible.
+
+
+        /// <summary>
+        /// Attempts to add a dash effect into the dash manager. It should be executed in a Load() function.
+        /// </summary>
+        /// <param name="dashEffect">An instance of the desired dash effect to add.</param>
+
+        public static void TryAddDash(PlayerDashEffect dashEffect)
+        {
+            //If DashIdentificationTable is not loaded or modder tries to load an abstract type of
+            //PlayerDashEffect, then don't add the dash and stop the method.
+            if (DashIdentificationTable == null || dashEffect.GetType().IsAbstract) return;
+
+            string id = (string)dashEffect.GetType().GetProperty("ID").GetValue(null);
+
+            //This chunk of the code only executes if the dash isn't already added and
+            //the dash id isn't empty.
+            if (!DashIdentificationTable.ContainsKey(id) && !String.IsNullOrEmpty(id))
```

Comment:
Under what situations would an ID collision like this arise? Is it safe/acceptable to just ignore it? Would it be better to unconditionally override the previous entry?

Reply:
Admittedly it is a bit of a "better safe than sorry" approach as to make sure that whomever is using the function wouldn't insert the same dash again and/or somehow used a dash that doesn't have an id (since the manager uses a dictionary to identify dashes after all)

Reply:
Would it then be better to override instead? I guess it makes more sense to remain UB with safety guards...

Reply:
I don't understand what would you mean by overriding, as all I could think of is that you would override the string id with something